frontend/src/modules/Dashboard/Components/Dashboard.tsx

```diff
@@ -38,11 +41,77 @@ export const Dashboard = () => {
       setGroups(res.data.data)
     })
   }, [])
+  // (a,b) = -1 if a before b, 1 if a after b, 0 if equal
+  function sorted(groups: CourseInfo[], menuValue: string) {
+    if (menuValue === 'newly-match-first') {
+      return [...groups].sort((a, _) => {
+        //-1 if a newly matchable and b isn't
+        if (a.lastGroupNumber === 0 && a.unmatched.length > 1) {
+          return -1
+        } else return 1
+      })
+    } else if (menuValue === 'unmatch-first') {
+      return [...groups].sort((a, _) => {
+        //-1 if a unmatchable and b isn't
+        if (a.lastGroupNumber === 0 && a.unmatched.length === 1) {
+          return -1
+        } else return 1
+      })
+    } else if (menuValue === 'match-first') {
+      return [...groups].sort((a, _) => {
+        //-1 if a matchable and b isn't
+        if (a.lastGroupNumber > 0 && a.unmatched.length > 0) {
+          return -1
+        } else return 1
+      })
+    } else if (menuValue === 'classesa-z') {
+      const sorted = [...groups].sort((a, b) => {
+        return a.names[0].localeCompare(b.names[0], undefined, {
+          numeric: true,
+        })
+      })
+      return sorted
+    } else if (menuValue === 'classesz-a') {
+      const sorted = [...groups].sort((a, b) => {
+        return b.names[0].localeCompare(a.names[0], undefined, {
+          numeric: true,
+        })
+      })
+      return sorted
+    } else return groups
+  }
+  const handleChange = (event: SelectChangeEvent) => {
+    setStatus(event.target.value)
+    setGroups(sorted(groups, event.target.value))
```

Comment:
Thought of a better way to do the sorting and not modify the original group array:
```js
const handleChange = (event: SelectChangeEvent) => {
  setStatus(event.target.value)
  // setGroups(sorted(groups, event.target.value))
}

const sortedGroups = sorted(groups, status)
```
Here the dropdown only changes the sort order, then every re-render a copy of the groups are sorted and used as `sortedGroups`, which is used for the groups component later. This makes the sorting possibly more predictable when there are ties because the previous sort of `groups` doesn't influence the next sort of `groups`, since `setGroups` is only called when loading the data.